src/liballoc/collections/vec_deque.rs

```diff
@@ -2172,6 +2172,14 @@ impl<'a, T> Iterator for Iter<'a, T> {
         accum = front.iter().fold(accum, &mut f);
         back.iter().fold(accum, &mut f)
     }
+
+    fn try_fold<B, F, R>(&mut self, init: B, mut f: F) -> R where
+        Self: Sized, F: FnMut(B, Self::Item) -> R, R: Try<Ok=B>
+    {
+        let (front, back) = RingSlices::ring_slices(self.ring, self.head, self.tail);
+        let accum = front.iter().try_fold(init, &mut f)?;
+        back.iter().try_fold(accum, &mut f)
```

Comment:
I'm concerned about this -- if it short-circuits it's not going to resume from the correct place, since it's `try_fold`ing on slice iterators and not updating the state of the `self` iterator.

Reply:
I'd like to see a test like this one:

https://github.com/rust-lang/rust/blob/7164a9f151a56316a382d8bc2b15ccf373e129ca/src/libcore/tests/iter.rs#L1990-L1993